internal/core/software_renderer.rs

```diff
@@ -1128,21 +1240,85 @@ impl<T: TargetPixel> RenderToBuffer<'_, T> {
         }
     }
 
+    fn foreach_region(
+        &mut self,
+        geometry: &PhysicalRect,
+        mut f: impl FnMut(&mut B, PhysicalRect, i16, i16),
+    ) {
+        let mut line = geometry.min_y();
+        while let Some(mut next) =
+            region_line_ranges(&self.dirty_region, line, &mut self.dirty_range_cache)
```

Comment:
region_line_ranges tries to split the ranges so that they span on lines. for the line by line rendering.
But here, i think it would be wiser to simply iterate over the rectangle in the region. I think they shouldn't overlap.

Example:
Imagine the following dirty region
```
+----+    +----+
|  A |    |  B |
|    |    +----+
|    |    
+----+
```

This will then generate a region for the upper part of A, B, and the lower part of A.
But it'd be better to "simply" draw A, then B.  Not?

Reply:
Sounds good to me.

Reply:
I hope I understood this correctly and applied this to rectangle and texture fills.

Reply:
I don't think you understood.
My point was that, in this funciton, instead of calling `region_line_ranges`,  just call self.dirty_region.iter_box(),  but the rectangles can overlap so that is not a good idea.

Reply:
I see. What do you suggest instead?